src/services/courseServices.js

```diff
@@ -0,0 +1,31 @@
+import axios from "../utils/axios";
+import { featchCourseRequest, featchCourseSuccess } from "../store/actions/courseActions";
+import errorHandler from "./errorHandler";
+
+const BASE_URL = "https://jsonplaceholder.typicode.com";
+
+export const fetchCourseData = () => {
+  return async (dispatch) => {
+    try {
+      dispatch(featchCourseRequest());
+      const response = await axios.get(`${BASE_URL}/posts`); // dummy URL for testing API call
+      const data = await response.data;
+      dispatch(featchCourseSuccess(data));
```

Comment:
Why are you dispatching `courseRequestFetch` twice?

Reply:
Please also correct spelling of `fetch`

Reply:
> Please also correct spelling of `fetch`

Sure @Abhishek-kumar09  I'll resolve that typo!

Reply:
> Why are you dispatching `courseRequestFetch` twice?

@Abhishek-kumar09   `fetchCourseRequest()` action is dispatch to set loading state `ture` once the user gets the Data  than another  `fetchCourseSuccess(data)` action is dispatch which set `loading` to  `false` and  store response data to `course:[ ]`.

As there are two functions one for  `course/all` and `course/:id`  in both the functions I have used the same `actions` for handling the states.

Reply:
Other things looks good to me.